lib/sprockets/cached_digest_utils.rb

```diff
@@ -0,0 +1,17 @@
+module Sprockets
+  module CachedDigestUtils
+    extend self
+
+    def cached_version_digest
+      version_string = defined?(version) ? version : ''
+
+      # Compute the initial digest using the implementation class. The
+      # Sprockets release version and custom environment version are
+      # mixed in. So any new releases will affect all your assets.
+      @cached_version_digest ||= digest_class.new.update(version_string.to_s)
+
+      # Returned a dupped copy so the caller can safely mutate it with `.update`
+      @cached_version_digest.dup
+    end
+  end
+end
```

Comment:
@jacobbednarz thanks for working on this. 

It's been a while since I looked at or thought about this, but I presume I was trying to follow the code patterns from previous releases as much as possible.

Let me know if I can answer any further questions or move this along in any way.